### DISCUSSION TITLE
Get original indices of selected nodes in Random Sampler

### DISCUSSION THREAD

**Discussion by itsdaniele** [pyg-team/pytorch_geometric#4532] [Q&A] [answered]
Hi there!

I was wondering if there is a simple way of getting the indices of the nodes that are selected for a given batch by a RandomSampler. 

I need that because I am trying to somehow have a connection between the full graph and the sampled subgraph.

Thanks a lot in advance.

**Comment by rusty1s** [comment, marked as answer]
```python
data.n_id = torch.arange(data.num_nodes)
loader = RandomNodeSampler(data, ...)
batch = next(iter(loader)
print(batch.n_id)
```
should do the trick. It is also documented [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.NeighborLoader).

**Reply by itsdaniele** [reply]
Thanks a lot, this works.
